ui/src/app/data/abstract-data.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from "@angular/core"
+import {HttpClient, HttpResponse} from "@angular/common/http"
```

Comment:
This is a new file, please make sure all imports are stylistically matching, i.e. with spaces before & after { } for single-line imports. Ideally grouped and alphabetized by import path.